crates/symbolicator/src/utils/futures.rs

```diff
@@ -235,6 +235,102 @@ where
     }
 }
 
+/// A guard to [`measure`] the amount of time it takes to download a source. This guard is also
```

Comment:
This was "heavily inspired" by `MeasureGuard` a few lines above in the same file. As far as I can tell, fairly invasive changes would need to be made to the existing `MeasureGuard` to work for the specific use case this PR needed. 

Some examples: I was unable to figure out how to adding an arbitrary number of tags to the metric it logs. It also doesn't support histograms out of the box. Throughput logging would have involved even more extensive changes to the struct. 

Given what was mentioned above, I opted to just use it as a loose template for a new type of `MeasureGuard` for source downloads.

Reply:
Done: I've moved `MeasureSourceDownload` to `services::download` and renamed it to an even wordier `MeasureSourceDownloadGuard`. Per your other suggestions the API has also been slimmed down to remove some of the unneeded flexibility inherited from the original `MeasureGuard` that isn't needed for this particular measure guard's use case.